arch/arm/core/cortex_m/prep_c.c

```diff
@@ -42,7 +42,7 @@ void __weak relocate_vector_table(void)
 #endif
 }
 
-#elif defined(CONFIG_ARMV7_M)
+#elif defined(CONFIG_ARMV7_M) || defined(CONFIG_ARMV8_M_MAINLINE)
```

Comment:
Maybe we could define some `ARMV7_V8_MAINLINE` to avoid having to `||` all the time?

Reply:
@carlescufi  Same answer as above: I am open to that option.